qa/integration-tests/src/test/java/io/zeebe/broker/it/network/AdvertisedAddressTest.java

```diff
@@ -0,0 +1,167 @@
+      final var messageSend =
+          client
+              .newPublishMessageCommand()
+              .messageName("test")
+              .correlationKey("test")
+              .send()
+              .join(5, TimeUnit.SECONDS);
+      // then - gateway can talk to the broker
+      assertThat(messageSend.getMessageKey()).isPositive();
+    }
+  }
+
+  private void configureBrokerContainer(final int index, final List<ZeebeContainer> brokers) {
+    final int clusterSize = brokers.size();
+    final var broker = brokers.get(index);
+    final var hostName = "broker-" + index;
+    final var commandApiProxy = toxiproxy.getProxy(hostName, ZeebePort.COMMAND.getPort());
+    final var internalApiProxy = toxiproxy.getProxy(hostName, ZeebePort.INTERNAL.getPort());
+    final var monitoringApiProxy = toxiproxy.getProxy(hostName, ZeebePort.MONITORING.getPort());
+
+    initialContactPoints.add(
+        TOXIPROXY_NETWORK_ALIAS + ":" + internalApiProxy.getOriginalProxyPort());
+
+    LoggerFactory.getLogger("Test")
+        .info("Configuring broker with initial contactpoints {}", initialContactPoints);
+
+    broker
+        .withNetwork(network)
+        .withNetworkAliases(hostName)
+        .withTopologyCheck(
+            new ZeebeTopologyWaitStrategy()
+                .forBrokersCount(clusterSize)
+                .forPartitionsCount(PARTITION_COUNT)
+                .forReplicationFactor(REPLICATION_FACTOR))
+        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "128KB")
+        .withEnv("ZEEBE_BROKER_CLUSTER_NODEID", String.valueOf(index))
+        .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERSIZE", String.valueOf(clusterSize))
+        .withEnv("ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR", String.valueOf(REPLICATION_FACTOR))
+        .withEnv("ZEEBE_BROKER_CLUSTER_PARTITIONCOUNT", String.valueOf(PARTITION_COUNT))
+        .withEnv(
+            "ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", String.join(",", initialContactPoints))
+        .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
+        .withEnv("ATOMIX_LOG_LEVEL", "INFO")
+        .withEnv("ZEEBE_BROKER_NETWORK_COMMANDAPI_ADVERTISEDHOST", TOXIPROXY_NETWORK_ALIAS)
+        .withEnv(
+            "ZEEBE_BROKER_NETWORK_COMMANDAPI_ADVERTISEDPORT",
+            String.valueOf(commandApiProxy.getOriginalProxyPort()))
+        .withEnv("ZEEBE_BROKER_NETWORK_INTERNALAPI_ADVERTISEDHOST", TOXIPROXY_NETWORK_ALIAS)
+        .withEnv(
+            "ZEEBE_BROKER_NETWORK_INTERNALAPI_ADVERTISEDPORT",
+            String.valueOf(internalApiProxy.getOriginalProxyPort()))
+        .withEnv("ZEEBE_BROKER_NETWORK_MONITORINGAPI_ADVERTISEDHOST", TOXIPROXY_NETWORK_ALIAS)
+        .withEnv(
+            "ZEEBE_BROKER_NETWORK_MONITORINGAPI_ADVERTISEDPORT",
+            String.valueOf(monitoringApiProxy.getOriginalProxyPort()))
+        // Since gossip does not work with ToxiProxy, increase the sync interval so changes are
+        // propagated faster
+        .withEnv("ZEEBE_BROKER_CLUSTER_MEMBERSHIP_SYNCINTERVAL", "1s");
```

Comment:
Any reason not to make it even faster? Or would it make no difference?

Reply:
No. I don't know if it makes a difference. We are not expecting any topology changes after the startup.